caput/tests/test_interferometry.py

```diff
@@ -0,0 +1,61 @@
+"""Test interferometry routines."""
+
+import pytest
+from math import pi, sqrt
+
+from caput import interferometry
+
+
+def test_sphdist():
+    from skyfield.units import Angle
+
+    # 90 degrees
+    assert interferometry.sphdist(
+        Angle(radians=0), Angle(radians=0), Angle(radians=pi / 2), Angle(radians=0)
+    ).radians == pytest.approx(pi / 2)
+    assert interferometry.sphdist(
+        Angle(radians=0), Angle(radians=0), Angle(radians=0), Angle(radians=pi / 2)
+    ).radians == pytest.approx(pi / 2)
+
+    # 60 degrees
+    assert interferometry.sphdist(
+        Angle(radians=0), Angle(radians=0), Angle(radians=pi / 4), Angle(radians=pi / 4)
+    ).radians == pytest.approx(pi / 3)
+    assert interferometry.sphdist(
+        Angle(radians=pi / 4), Angle(radians=pi / 4), Angle(radians=0), Angle(radians=0)
+    ).radians == pytest.approx(pi / 3)
+
+
+def test_rotate_ypr():
+
+    # No rotation
+    basis = interferometry.rotate_ypr([0, 0, 0], 1, 2, 3)
+    assert basis == pytest.approx([1, 2, 3])
+
+    # Rotate into +Y two ways
+    basis = interferometry.rotate_ypr([pi / 2, 0, 0], 1, 0, 0)
+    assert basis == pytest.approx([0, 1, 0])
+
+    basis = interferometry.rotate_ypr([0, pi / 2, 0], 0, 0, 1)
+    assert basis == pytest.approx([0, 1, 0])
+
+    # General rotation
+    x0 = sqrt(2) / 2
+    y0 = 0.5
+    z0 = 0.5
+    basis = interferometry.rotate_ypr([pi / 2, pi / 3, pi / 4], x0, y0, z0)
+
+    # The calculation goes like this:
+
+    # After yawing by 90 degrees:
+    # x1 = -y0               y1 = x0                     z1 = z0
+
+    # Then pitching by 60 degrees:
+    # x2 = x1                y2 = 0.5 * y1 + v3/2 z1     z2 = -v3)/2 y1 + 0.5 z1
```

Comment:
I guess `v2` and `v3` refer to sqrt(2) and sqrt(3)? For consistency, maybe remove the `)` after the second `v3`?